From: Kalesh AP <kalesh.purayil@emulex.com>
Date: Thu, 17 Jul 2014 16:20:20 +0530
Subject: [PATCH 040/145] be2net: fix error status for FW-download
Patch-mainline: v3.17-rc1
Git-commit: 56ace3a0d0209ea9f1806abb9c1046fdb89e0030
References: bsc#921784 FATE#318561

For FW download ethtool cmd, if the user provides an FW-image incompatible
with the chip, return -EINVAL instead of -1.

Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/emulex/benet/be_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -3952,7 +3952,7 @@ static int be_flash_skyhawk(struct be_ad
 	fsec = get_fsec_info(adapter, filehdr_size + img_hdrs_size, fw);
 	if (!fsec) {
 		dev_err(dev, "Invalid Cookie. FW image may be corrupted\n");
-		return -1;
+		return -EINVAL;
 	}
 
 	for (i = 0; i < le32_to_cpu(fsec->fsec_hdr.num_images); i++) {
